### PR TITLE
Request spec

### PR DESCRIPTION
**Pull request by LuisMenjivar**

### CODE REVIEW

File: spec/requests/create_bookmark_from_email_spec.rb [hunk 1]
```diff
@@ -13,12 +13,23 @@
     expect(@user.topics.first.bookmarks).to match_array [Bookmark.find_by(url: "www.google.com")]
     expect(response.status).to eq(200) 
   end
-  xit "creates a bookmark if topic already exists" do
+  it "creates a bookmark if topic already exists" do
+    topic = create(:topic, title: "animals", user: @user)
+    post '/bookmarks',{sender: "test@example.com", subject: "animals", "body-plain": "www.google.com"}
+    expect(@user.topics.first.bookmarks.first).to belong_to(:topic)
```
Comment (eliotsykes) [inline spec/requests/create_bookmark_from_email_spec.rb:19]: I think the `belong_to(:topic)` test here isn't testing what's intended. Check for yourself by commenting out line 17. The test will still pass without line 17 I suspect.

File: spec/requests/create_bookmark_from_email_spec.rb [hunk 2]
```diff
@@ -13,12 +13,23 @@
...
   end
-  xit "does not create a topic nor a bookmark if user does not exists" do 
+  it "does not create a topic nor a bookmark if user does not exists", :skip_before do
+    post '/bookmarks',{sender: "test@example.com", subject: "animals", "body-plain": "www.google.com"}
```
Comment (eliotsykes) [inline spec/requests/create_bookmark_from_email_spec.rb:22]: Conventional to put a space after the comma `'/bookmarks', {`

File: spec/requests/create_bookmark_from_email_spec.rb [hunk 3]
```diff
@@ -13,12 +13,23 @@
...
+    expect(Topic.all.count).to eq(0)
+    expect(Bookmark.all.count).to eq(0)
```
Comment (eliotsykes) [inline spec/requests/create_bookmark_from_email_spec.rb:24]: `Topic.all.count` can be shortened to `Topic.count`. Likewise on line 24 for `Bookmark.count`

File: spec/requests/create_bookmark_from_email_spec.rb [hunk 4]
```diff
@@ -13,12 +13,23 @@
...
+    expect(response.status).to eq(200) 
   end
-  xit "does not create a bookmark without a topic" do 
+  it "does not create a bookmark without a topic" do 
+    expect{post '/bookmarks',{sender: "test@example.com", subject: "", "body-plain": "www.google.com"}
+       }.to raise_error(ActiveRecord::RecordInvalid)
```
Comment (eliotsykes) [inline spec/requests/create_bookmark_from_email_spec.rb:29]: Reformat this `it` block and the `it` block below to read a bit clearer:

``` ruby
expect do
  post '/bookmarks', { PARAMS GO HERE }
end.to raise_error(ActiveRecord::RecordInvalid)
```

In addition, add a check to the end of each `it` block that no bookmarks and/or topics have been created. Currently the tests would pass if the bookmark was created before the error is raised by the controller.

File: app/models/topic.rb [hunk 5]
```diff
@@ -1,4 +1,5 @@
 class Topic < ActiveRecord::Base
   belongs_to :user
   has_many :bookmarks
+  validates :title, presence: true
```
Comment (eliotsykes) [inline app/models/topic.rb:4]: Add validation tests to the model specs for `Topic` and for `Bookmark`

File: app/controllers/bookmarks_controller.rb [hunk 6]
```diff
@@ -9,7 +9,7 @@ def create_bookmark_from_mail
        url = params["body-plain"]
       if topic.nil?
         new_topic = Topic.create!(title: "#{params[:subject]}", user: user)
-        Bookmark.create(url: "#{url}", topic: new_topic)
+        Bookmark.create!(url: "#{url}", topic: new_topic)
```
Comment (eliotsykes) [inline app/controllers/bookmarks_controller.rb:12]: Try to clean up so the string interpolation isn't needed for `url`

File: app/controllers/bookmarks_controller.rb [hunk 7]
```diff
@@ -9,7 +9,7 @@ def create_bookmark_from_mail
...
       else
         Bookmark.create(url: "#{url}", topic: topic)
       end
```
Comment (eliotsykes) [inline app/controllers/bookmarks_controller.rb:15]: We can simplify these few lines from:

``` ruby
      if topic.nil?
        new_topic = Topic.create!(title: "#{params[:subject]}", user: user)
        Bookmark.create!(url: "#{url}", topic: new_topic)
      else
        Bookmark.create(url: "#{url}", topic: topic)
      end
```

to:

``` ruby
      if topic.nil?
        topic = Topic.create!(title: "#{params[:subject]}", user: user)
      end
      Bookmark.create(url: "#{url}", topic: topic)
```

File: spec/requests/create_bookmark_from_email_spec.rb (hunk 1 above)
Comment (LuisMenjivar) [inline spec/requests/create_bookmark_from_email_spec.rb:19]: You were right, the test passes when I commented out line 17. I changed it
to : expect(Bookmark.find_by_url("www.google.com").topic).to eq(topic)
and added :  expect(@user.topics.count).to eq(1)
before and after the post request.

**Pull request merged**